### ISSUE TITLE
Add ability to set wireguard IP via node label

### ISSUE THREAD

**Issue by jawabuu** [squat/kilo#221]
In cases where one needs prior knowledge of the nodes IP to for example;
1. Create host entries that resolve hostnames to kilo created wireguard IPs
A node label that can be set on cluster creation could be implemented to achieve this.
Suggesting to use a label as for most k8s distros, this can be set on node/cluster creation.
This label should indicate a `preferred` wireguard IP to be set by kilo and ignored if another node already exists with the IP.
This means all other functionality can stay the same.
So in short, 
1. Check Node Label e.g. `kilo.squat.ai/wireguard-ip:"<IPV4 value>"`
2. If label exists, check if IP is valid for the kilo subnet and is available.
3. Use the IP for the kilo interface
@squat

**Comment by jawabuu** [comment]
This is mostly useful for clsters using kilo as CNI

**Comment by squat** [comment]
Kilo uses annotations on the node objects to override and force values that would otherwise be autonomously determined. Let's align this proposal with that practice, e.g. `kilo.squat.ai/force-wireguard-ip`. I think we should likely use the `force-` prefix to align with the current practices within Kilo. More importantly, it allows kilo to understand the provenance of a value. Otherwise it will cause problems as nodes do not know if the value is out of date and needs to be updated as the cluster topology has changed or if the value was explicitly set by the administrator.

**Comment by Miras4207** [comment]
I stumbled upon this question myself and wonder if there is any progress on this, of if there is any other way to do this.

In my setup, I have only a control node with public IP, then a couple of worker nodes behind NAT. Currently, I am even forced to set [`kilo.squat.ai/force-internal-ip`](https://kilo.squat.ai/docs/annotations#force-internal-ip) on worker nodes to WireGuard IP to achieve full connectivity. Being able to set some fixed WireGuard IP would be helpful.

**Comment by markusressel** [comment]
I am successfully using the annotation `kilo.squat.ai/force-endpoint=192.168.2.10:51820` to set the wireguard IP for a node. In my case I need to avoid the default which is using an IPv6 address and set an IPv4 address manually. I have also set the [kilo.squat.ai/force-internal-ip](https://kilo.squat.ai/docs/annotations#force-internal-ip) annotation mentioned by @Miras4207 .
